apps/daimo-mobile/src/view/screen/OnboardingScreen.tsx

```diff
@@ -582,25 +602,179 @@ function UseExistingPage({
   onPrev?: () => void;
   daimoChain: DaimoChain;
 }) {
+  // Use existing account spin loops and waits for this device to show up
+  // in some on-chain account.
   const { status, message, pubKeyHex } = useExistingAccount(daimoChain);
 
   useEffect(() => {
     if (status === "success") onNext();
   }, [status]);
 
+  // Account and associated logic used to restore from backup passkey ephemerally.
+  const [passkeyAccount, setPasskeyAccount] = useState<Account | undefined>(
+    undefined
+  );
+
+  const nextSlot = useMemo(
+    () =>
+      passkeyAccount
+        ? findUnusedSlot(
+            passkeyAccount.accountKeys.map((k) => k.slot),
+            env(daimoChain).deviceType === "phone"
+              ? SlotType.Phone
+              : SlotType.Computer
+          )
+        : null,
+    [passkeyAccount]
+  );
+
+  const nonce = useMemo(
+    () => new DaimoNonce(new DaimoNonceMetadata(DaimoNonceType.AddKey)),
+    [pubKeyHex]
+  );
+
+  const sendFn = async (opSender: DaimoOpSender) => {
+    if (!pubKeyHex || !passkeyAccount || nextSlot === null)
+      throw new Error("[ACTION] passkey account not ready");
+    console.log(`[ACTION] adding device ${pubKeyHex}`);
+    return opSender.addSigningKey(nextSlot, pubKeyHex, {
+      nonce,
+      chainGasConstants: passkeyAccount.chainGasConstants,
+    });
+  };
+
+  const {
+    status: addDeviceStatus,
+    message: addDeviceMessage,
+    exec: addDeviceExec,
+  } = useSendAsync({
+    dollarsToSend: 0,
+    sendFn,
+    passkeyAccount,
+  });
+
+  const [restoreStatus, setRestoreStatus] = useState<{
+    status: ActStatus;
+    message: string;
+  }>({ status: "idle", message: "Restore from backup" });
+
+  const onRestoreBackup = async () => {
+    assert(pubKeyHex !== undefined);
+    console.log(`[ONBOARDING] restore backup attempt`);
+    setRestoreStatus({ status: "loading", message: "Requesting backup" });
+
+    const bChallenge = hexToBytes("0xdead");
+    const challengeB64 = base64.encode(bChallenge);
```

Comment:
recommend moving logic out of `OnboardingScreen`

Reply:
moved to new folder / components